iceoryx_posh/test/moduletests/test_posh_runtime.cpp

```diff
@@ -0,0 +1,339 @@
+    virtual ~PoshRuntime_test()
+    {
+    }
+
+    virtual void SetUp(){};
+    virtual void TearDown(){};
+
+    void InterOpWait()
+    {
+        std::this_thread::sleep_for(std::chrono::milliseconds(200));
+    }
+
+    RouDiEnvironment m_roudiEnv{iox::RouDiConfig_t().setDefaults()};
+    PoshRuntime* m_runtime{&iox::runtime::PoshRuntime::getInstance("/sender")};
+    MqMessage m_sendBuffer;
+    MqMessage m_receiveBuffer;
+    const iox::cxx::string<100> m_runnableName{"testRunnable"};
+    const iox::cxx::string<100> m_invalidRunnableName{"invalidRunnable,"};
+    static bool m_errorHandlerCalled;
+    const iox::cxx::string<100> m_runtimeName{"/sender"};
+};
+
+bool PoshRuntime_test::m_errorHandlerCalled{false};
+
+
+TEST_F(PoshRuntime_test, ValidAppname)
+{
+    std::string appName("/valid_name");
+
+    EXPECT_NO_FATAL_FAILURE({ PoshRuntime::getInstance(appName); });
+}
+
+
+TEST_F(PoshRuntime_test, AppnameLength_OutOfLimit)
+{
+    std::string tooLongName(100, 's');
+    tooLongName.insert(0, 1, '/');
+
+    EXPECT_DEATH({ PoshRuntime::getInstance(tooLongName); },
+                 "Application name has more than 100 characters, including null termination!");
+}
+
+
+TEST_F(PoshRuntime_test, MaxAppnameLength)
+{
+    std::string maxValidName(99, 's');
+    maxValidName.insert(0, 1, '/');
+
+    EXPECT_NO_FATAL_FAILURE({ PoshRuntime::getInstance(maxValidName); });
+}
+
+
+TEST_F(PoshRuntime_test, NoAppname)
+{
+    const std::string invalidAppName("");
+
+    EXPECT_DEATH({ PoshRuntime::getInstance(invalidAppName); },
+                 "Cannot initialize runtime. Application name must not be empty!");
+}
+
+
+TEST_F(PoshRuntime_test, NoLeadingSlash_Appname)
+{
+    const std::string invalidAppName = "invalidname";
+
+    EXPECT_DEATH(
+        { PoshRuntime::getInstance(invalidAppName); },
+        "Cannot initialize runtime. Application name invalidname does not have the required leading slash '/'");
+}
+
+
+// since getInstance is a singleton and test class creates instance of Poshruntime,
+// when getInstance() is called without parameterx it reuturns extisting instance
+// To be able to test this, it needs to be the very first call to getInstance but since,
+// we have multiple tests in this binary its not possible here to test
+TEST_F(PoshRuntime_test, DISABLED_AppnameEmpty)
+{
+    EXPECT_DEATH({ iox::runtime::PoshRuntime::getInstance(); },
+                 "Cannot initialize runtime. Application name has not been specified!");
+}
+
+
+TEST_F(PoshRuntime_test, GetInstanceName_ReturnValue)
+{
+    const std::string appname = "/app";
+
+    auto& sut = PoshRuntime::getInstance(appname);
+
+    EXPECT_EQ(sut.getInstanceName(), appname);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareApplicationIsSuccessful)
+{
+    uint32_t uniqueIdCounter = iox::popo::BasePortData::s_uniqueIdCounter;
+
+    const auto applicationPortData = m_runtime->getMiddlewareApplication();
+
+    ASSERT_THAT(m_runtimeName, applicationPortData->m_processName);
```

Comment:
this is not what I meant. You need to check that `applicationPortData` is not a `nullptr` before you access `m_processName`

Reply:
This nullptr check is still missing.